tests/test_opencode_rust_coverage_toolchain_contract.py

```
@@ -0,0 +1,69 @@
+
+    helper = _helper_text()
+    assert _LLVM_COV_PATH in helper
+    assert _LLVM_PROFDATA_PATH in helper
+    assert "unversioned" not in helper
+
+
+def test_isolated_runtime_revalidates_llvm_tools_before_coverage() -> None:
+    """Require reviewed-path equality and executable checks before coverage."""
+
+    helper = _helper_text()
+    assert f'"${{LLVM_COV:-}}" != "$LLVM_COV_PATH"' in helper
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)